src/lib.rs

```diff
@@ -1402,17 +1404,22 @@ impl<K: KVStore + Sync + Send + 'static> Node<K> {
 				continue;
 			}
 
-			if let Some(first_route_hop) = path.hops.first() {
-				if let Some(first_hop) = first_hops
-					.iter()
-					.find(|h| h.short_channel_id == Some(first_route_hop.short_channel_id))
-				{
+			if let Some(first_path_hop) = path.hops.first() {
+				if let Some(first_hop) = first_hops.iter().find(|h| {
+					h.get_outbound_payment_scid() == Some(first_path_hop.short_channel_id)
+				}) {
 					let path_value = path.final_value_msat() + path.fee_msat();
+					let used_liquidity =
+						used_liquidity_map.entry(first_path_hop.short_channel_id).or_insert(0);
+
 					if first_hop.next_outbound_htlc_limit_msat
-						< path_value * self.config.probing_liquidity_limit_multiplier
+						< (*used_liquidity + path_value)
+							* self.config.probing_liquidity_limit_multiplier
```

Comment:
Isn't `next_outbound_htlc_limit_msat` reduced on each probe? So you would potentially stop sending probes even though `1 / probing_liquidity_limit_multiplier` has not been exhausted for the given *payment*? (i.e., a smaller limit is being compared against each path rather than being fixed).

Reply:
Maybe we don't care, but just wanted to mention that in case it was not intentional.

Reply:
Ah, right. I forgot that those are copies of `Channel` data.